api/include/opentelemetry/trace/tracer-factory.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <vector>
+#include <memory>
+#include <string>
+
+#include "opentelemetry/nostd/string_view.h"
+
+#include "./tracer.h"
+
+using opentelemetry::nostd::string_view;
+
+namespace opentelemetry
+{
+namespace trace
+{
+class Tracer;
+class TracerFactory
```

Comment:
This class should provide a virtual method `getTracer` to obtain a `Tracer` instance. The tracer "SDK" API consists of this factory. The user will construct and configure a SDK and then return one of these factories to the user. The user can use this directly, or it can be installed as the global instance, so the static `getInstance` here should be clearly marked as a global. In the Golang SDK, we placed global state in a separate package to clearly separate them from the Tracer and Meter APIs.

Reply:
I'm still a bit new to the conversations around this stuff, so thanks for detailing all that! I've moved the `getInstance` to `opentelemetry::global::Factory::getTracerFactory` to be more explicit, and moved the actual tracer creation and tracking to the SDK side of things.